benchalerts/benchalerts/pipeline_steps/github.py

```diff
@@ -230,6 +235,82 @@ def _default_status_description(full_comparison: FullComparisonInfo) -> str:
             )
 
 
+class GitHubPRCommentAboutCheckStep(AlertPipelineStep):
+    """An ``AlertPipeline`` step to make a comment on a PR about a GitHub Check that was
+    created by a previously-run ``GitHubCheckStep``. This is useful if you're running
+    benchmarks on a merge-commit, and no one is necessarily monitoring the Checks on the
+    default branch. It should be set up to notify the PR that caused the merge-commit,
+    so that the relevant people can take action if necessary.
+
+    Parameters
+    ----------
+    pr_number
+        The number of the PR to make the comment on.
+    repo
+        The repo name to make the comment on, in the form "owner/repo". Either provide
+        this or ``github_client``.
+    github_client
+        A GitHubRepoClient instance. Either provide this or ``repo``.
+    check_step_name
+        The name of the ``GitHubCheckStep`` that ran earlier in the pipeline. Defaults
+        to "GitHubCheckStep" (which was the default if no name was given to that step).
+    step_name
+        The name for this step. If not given, will default to this class's name.
+
+    Returns
+    -------
+    dict
+        The response body from the GitHub HTTP API as a dict.
+
+    Notes
+    -----
+    Environment variables
+    ~~~~~~~~~~~~~~~~~~~~~
+    ``GITHUB_APP_ID``
+        The ID of a GitHub App that has been set up according to this package's
+        instructions and installed to your repo. Recommended over ``GITHUB_API_TOKEN``.
+        Only required if ``repo`` is provided instead of ``github_client``.
+    ``GITHUB_APP_PRIVATE_KEY``
+        The private key file contents of a GitHub App that has been set up according to
+        this package's instructions and installed to your repo. Recommended over
+        ``GITHUB_API_TOKEN``. Only required if ``repo`` is provided instead of
+        ``github_client``.
+    ``GITHUB_API_TOKEN``
+        A GitHub Personal Access Token with the ``repo:status`` permission. Only
+        required if not authenticating with a GitHub App, and if ``repo`` is provided
+        instead of ``github_client``.
+    """
+
+    def __init__(
+        self,
+        pr_number: int,
```

Comment:
I was thinking about how we might "automatically" figure this out.

The GitHub API does have a ["get PRs associated with a given commit"](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit) endpoint, which does work on repositories like `apache/arrow`. For a certain default-branch commit (which is given in the alert pipeline input), we would be able to use this endpoint to comment on the PR that caused the commit. That's nice.

However it doesn't work for repositories like `facebookincubator/velox`, because they have a bot [close their PRs](https://github.com/facebookincubator/velox/pull/4297#event-8758907106) instead of going through the GitHub merge process. So that endpoint always returns an empty list. But (check out that link) I noticed the commit is still associated with that "closing" event in the UI. Given that information (and [this](https://stackoverflow.com/a/49327668) sort-of-helpful SO answer), I was able to construct a workaround GitHub GraphQL API query that maps each Velox PR to its merge-commit:
```
{
  repository(name: "velox", owner: "facebookincubator") {
    pullRequests(last: 10, states: [CLOSED]) {
      edges {
        node {
          number
          timelineItems(last: 100, itemTypes: CLOSED_EVENT) {
            edges {
              node {
                ... on ClosedEvent {
                  closer {
                    ... on Commit {
                      oid
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```
Try it out in the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer). Given a merge-commit, we could use this query (with careful limits) to find the respective pull request. However, this seems like a lot of complexity to carry around for an uncommon use case. But I'm not sure of another way to automatically find this mapping. I suppose we could parse the commit message, but that seems brittle and wrong.

So after all that, I decided to have the person implementing `benchalerts` figure out how to provide the pull number. I'll try to ensure this is possible for all the Conbench instances we know about. And maybe we can add in "smart" functionality later.

Reply:
Maybe we should've stored PR number on `Commit` instead of the branch...